src/Microsoft.AspNet.Http.Abstractions/Extensions/UseMiddlewareExtensions.cs

```diff
@@ -23,23 +24,42 @@ public static IApplicationBuilder UseMiddleware(this IApplicationBuilder builder
             return builder.Use(next =>
             {
                 var instance = ActivatorUtilities.CreateInstance(builder.ApplicationServices, middleware, new[] { next }.Concat(args).ToArray());
-                var methodinfo = middleware.GetMethod("Invoke", BindingFlags.Instance | BindingFlags.Public);
+                var methods = middleware.GetMethods(BindingFlags.Instance | BindingFlags.Public);
+                if (methods.Count(m => String.Equals(m.Name, "Invoke", StringComparison.Ordinal)) > 1)
+                {
+                    throw new InvalidOperationException(Resources.FormatException_UseMiddleMutlipleInvokes("Invoke"));
```

Comment:
Naah I actually prefer an InvalidOperation because even though *we're* using Reflection, the error is not itself a Reflection error - it's a particular design pattern that we require.